host/class/uvc/usb_host_uvc_2/include/usb/uvc_host.h

```diff
@@ -0,0 +1,241 @@
+        uint32_t frame_heap_caps;    /**< Memory capabilities for frame buffers. Directly passed to heap_caps_malloc() */
+        int number_of_urbs;          /**< Number of URBs for this stream. Triple buffering scheme is recommended */
+        size_t urb_size;             /**< Size in bytes of 1 URB, 10kB should be enough for start.
+                                          Larger value results in less frequent interrupts at the cost of memory consumption */
+    } advanced;
+} uvc_host_stream_config_t;
+
+/**
+ * @brief Install UVC driver
+ *
+ * - USB Host Library must already be installed before calling this function (via usb_host_install())
+ * - This function should be called before calling any other UVC driver functions
+ *
+ * @param[in] driver_config Driver configuration structure. If set to NULL, a default configuration will be used.
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_install(const uvc_host_driver_config_t *driver_config);
+
+/**
+ * @brief Uninstall UVC driver
+ *
+ * - User must ensure that all UVC devices must be closed via uvc_host_stream_close() before calling this function
+ *
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_uninstall(void);
+
+/**
+ * @brief Open UVC compliant device
+ *
+ * @param[in]  stream_config  Configuration structure of the device
+ * @param[in]  timeout        Timeout in FreeRTOS ticks
+ * @param[out] stream_hdl_ret UVC stream handle
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_stream_open(const uvc_host_stream_config_t *stream_config, int timeout, uvc_host_stream_hdl_t *stream_hdl_ret);
+
+/**
+ * @brief Start UVC stream
+ *
+ * After this call, the user will be informed about new frames by frame callback
+ *
+ * @param[in] stream_hdl UVC handle obtained from uvc_host_stream_open()
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_stream_start(uvc_host_stream_hdl_t stream_hdl);
+
+/**
+ * @brief Stop UVC stream
+ *
+ * @param[in] stream_hdl UVC handle obtained from uvc_host_stream_open()
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_stream_stop(uvc_host_stream_hdl_t stream_hdl);
+
+/**
+ * @brief Pause UVC stream
+ *
+ * After this call, the USB transfers are not re-submitted and the user will stop getting new frame callbacks.
+ * This function does not issue any CTRL request. In only stops receiving new data from streaming endpoint.
+ *
+ * @param[in] stream_hdl UVC handle obtained from uvc_host_stream_open()
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_stream_pause(uvc_host_stream_hdl_t stream_hdl);
+
+/**
+ * @brief Start UVC stream
+ *
+ * After this call, the user will be informed about new frames by frame callback.
+ * This function does not issue any CTRL request. In only starts receiving new data from streaming endpoint.
+ *
+ * @param[in] stream_hdl UVC handle obtained from uvc_host_stream_open()
+ * @return esp_err_t
+ */
+esp_err_t uvc_host_stream_unpause(uvc_host_stream_hdl_t stream_hdl);
```

Comment:
I don't get the idea of `uvc_host_stream_unpause` as a separate function. 
Could it be merged with `uvc_host_stream_start`? 

And can we just merge them and keep only `uvc_host_stream_start` and `uvc_host_stream_stop`?

Reply:
You're right, this functions were added when we're experimenting with the channel HALT bug on.

I'll clean this up